share/spice/just_delete_me/just_delete_me.js

```diff
@@ -0,0 +1,69 @@
+(function (env) {
+    "use strict";
+    env.ddg_spice_just_delete_me = function(api_result) {
+        if (!api_result || api_result.length < 1) {
+            return Spice.failed('just_delete_me');
+        }
+        var script = $('[src*="/js/spice/just_delete_me/"]')[0],
+            source = $(script).attr("src"),
+            query = source.match(/just_delete_me\/([^\/]+)/)[1],
+            decodedQuery = decodeURIComponent(query).split(" ")[0].toLowerCase();
+
+        if (!decodedQuery || decodedQuery.length < 4) {
+            return Spice.failed('just_delete_me');
+        }
+
+        var exact = api_result.filter(function(item) { return item.name.toLowerCase() === decodedQuery}),
+            exact = exact.length > 0 ? [exact[0]] : 
+                    api_result.filter(function(item) { return item.domains &&
```

Comment:
As I mentioned before, the DDG.isRelevant() and DDG.stringsRelevant() does not really return what is expected here, returning too many false positives IMO.

The result for your query on iCloud is correct. There is no entry for iCloud on JustDeleteMe, though there is one for JoliCloud as you noted, which matches. See http://justdelete.me/#icloud